examples/sample-cmd/main.go

```diff
@@ -23,6 +25,43 @@ func main() {
 		return fmt.Sprintf("Hello %s!", c.Param("name")), nil
 	})
 
+	app.SubCommand("spinner", func(ctx *gofr.Context) (interface{}, error) {
+		// initialize the spinner
+		spinner := terminal.NewDotSpinner(ctx.Out).Spin(ctx).Spin(ctx)
+
+		select {
+		case <-ctx.Done():
+			spinner.Stop()
+			return nil, ctx.Err()
+		case <-time.After(2 * time.Second):
+			spinner.Stop()
+		}
```

Comment:
I would use this,it would prevent people to forget the call to Stop if a new `case` is added

```suggestion
		defer spinner.Stop()
		select {
		case <-ctx.Done():
			return nil, ctx.Err()
		case <-time.After(2 * time.Second):
		}
```